src/controller/CommissioningDelegate.h

```diff
@@ -370,9 +370,12 @@ class CommissioningParameters
         mAttestationNonce.SetValue(attestationNonce);
         return *this;
     }
+
+    // If a WiFiCredentials is provided, then the WiFiNetworkScan will not be attempted
     CommissioningParameters & SetWiFiCredentials(WiFiCredentials wifiCreds)
     {
         mWiFiCreds.SetValue(wifiCreds);
+        mAttemptWiFiNetworkScan = MakeOptional(static_cast<bool>(false));
```

Comment:
I know you copied the Thread code, but this seems like a complicated way to do:
```suggestion
        mAttemptWiFiNetworkScan.SetValue(false);
```